packages/teams-js/src/public/dialog.ts

```diff
@@ -210,6 +217,121 @@ export namespace dialog {
     }
   }
 
+  /**
+   * @hidden
+   * Hide from docs
+   * --------
+   * Convert AdaptiveCardDialogInfo to DialogInfo to send the information to host in {@linkcode adaptiveCard.open} API.
+   *
+   * @internal
+   */
+  export function getDialogInfoFromAdaptiveCardDialogInfo(adaptiveCardDialogInfo: AdaptiveCardDialogInfo): DialogInfo {
+    const dialogInfo: DialogInfo = {
+      card: adaptiveCardDialogInfo.card,
+      height: adaptiveCardDialogInfo.size ? adaptiveCardDialogInfo.size.height : DialogDimension.Small,
+      width: adaptiveCardDialogInfo.size ? adaptiveCardDialogInfo.size.width : DialogDimension.Small,
+      title: adaptiveCardDialogInfo.title,
+    };
+    return dialogInfo;
+  }
+
+  /**
+   * @hidden
+   * Hide from docs
+   * --------
+   * Convert BotAdaptiveCardDialogInfo to DialogInfo to send the information to host in {@linkcode adaptiveCard.open} API.
+   *
+   * @internal
+   */
+  export function getDialogInfoFromBotAdaptiveCardDialogInfo(
+    botAdaptiveCardDialogInfo: BotAdaptiveCardDialogInfo,
+  ): DialogInfo {
+    const dialogInfo: DialogInfo = {
+      card: botAdaptiveCardDialogInfo.card,
+      height: botAdaptiveCardDialogInfo.size ? botAdaptiveCardDialogInfo.size.height : DialogDimension.Small,
+      width: botAdaptiveCardDialogInfo.size ? botAdaptiveCardDialogInfo.size.width : DialogDimension.Small,
+      title: botAdaptiveCardDialogInfo.title,
+      completionBotId: botAdaptiveCardDialogInfo.completionBotId,
+    };
+    return dialogInfo;
+  }
+  /**
+   * Subcapability for interacting with adaptive card dialogs
+   */
+  export namespace adaptiveCard {
```

Comment:
should we add beta tag to this?

Reply:
Just wondering why, we want to state it as `beta`?

Reply:
In case you had to tweak it a little in future (based on customer feedback or you find a bug), you will be able to change it without having to bump the major version of library